editor/src/components/editor/editor-component.tsx

```diff
@@ -484,3 +486,52 @@ const ToastRenderer = React.memo(() => {
     </FlexColumn>
   )
 })
+
+const LockedOverlay = React.memo(() => {
+  const leftMenuExpanded = useEditorState(
+    (store) => store.editor.leftMenu.expanded,
+    'EditorComponentInner leftMenuExpanded',
+  )
+
+  const editorLocked = useEditorState(
+    (store) => store.editor.githubOperations.length > 0,
+    'EditorComponentInner editorLocked',
+  )
+
+  const anim = keyframes`
+    from {
+      opacity: 0;
+    }
+    to {
+      opacity: 0.2;
+    }
+  `
+
+  if (!editorLocked) {
+    return null
+  }
+
+  return (
+    <div
+      onMouseDown={(e) => e.preventDefault()}
```

Comment:
We could have a single `React.useCallback` or separate top level function that we use for all of these

Reply:
https://github.com/concrete-utopia/utopia/pull/2707/commits/1b5966e3df89ab4b5a029c1248953fe85b7abb69